modules/server/src/network/aggregations/index.ts

```diff
@@ -172,3 +168,31 @@ const resolveNumericAggregation = (aggregations: NumericAggregations) => {
 	// TODO: implement
 	throw Error('Not implemented');
 };
+
+const emptyAggregation: NetworkAggregation = { bucket_count: 0, buckets: [] };
+
+export class AggregationAccumulator {
+	totalAgg: any; // TODO: Fix any typing
+
+	constructor(requestedFields: any) {
+		/*
+		 * seed accumulator with the requested field keys
+		 * this will make it easier to add to using key lookup instead of Array.find
+		 */
+		this.totalAgg = Object.keys(requestedFields).reduce((accumulator: any, field: any) => {
+			return { ...accumulator, [field]: emptyAggregation };
+		}, {});
+	}
+
+	resolve(data, fields) {
+		resolveAggregations({
+			accumulator: this.totalAgg,
+			networkResult: data,
+			requestedAggregationFields: fields,
+		});
```

Comment:
Why wrap another function here... could just move the contents of resolveAggregation to be defined inside this class...

Also, I can't approve a function without argument type declarations. Right now you could do `.resolve(true, false)` and TS will be happy with you.

Reply:
`resolveAggregation` was created already and it works nice, if moved into this class it has to be tied to the class and not an isolated function because it references class properties.

Reply:
tied to module scope already so I'm worried about visibility outside the class